File: PHP/7php 3/orderedCount.php
<?php
/*
|--------------------------------------------------------------------------
| Ordered Count of Characters
|--------------------------------------------------------------------------
|
*/

function orderedCount($text) {
   $res = [];
   foreach (str_split($text) as $v) {
      $res[$v] = [$v, ($res[$v][1] ?? 0) + 1];
   }
   return strlen($text) ? array_values($res) : [];
}


function orderedCount($text) {
   foreach(array_count_values(str_split($text)) as $char => $count) {
      $char !== '' ? $result[] = [$char, $count] : $result = [];
   }
   return $result;
}


function orderedCount($text) {
   $array = [];
   foreach(array_count_values(str_split($text)) as $letter => $count) {
      $array[] = [$letter, $count];
   }
   return empty($text) ? [] : $array;
}


print_r(orderedCount("abracadabra"));

File: Python/7py 2/ordered_count.py
"""
|--------------------------------------------------------------------------
| Ordered Count of Characters
|--------------------------------------------------------------------------
|
"""


from collections import Counter
def ordered_count(inp):
   return list(Counter(inp).items())


def ordered_count(inp):
   return [(i, inp.count(i)) for i in sorted(set(inp), key=inp.index)]


def ordered_count(inp):
   return [(x, inp.count(x)) for x in list(dict.fromkeys(inp))]
   

print(ordered_count("abracadabra"), [
   ["a", 5],
   ["b", 2],
   ["r", 2],
   ["c", 1],
   ["d", 1],
])